### PR TITLE
Adds a new option to gallery format - lightgallery

### PR DESCRIPTION
**Pull request by nischayn22**
Similar to fancybox option this change adds lightgallery as another
option.

See https://sachinchoolur.github.io/lightGallery/demos/

### CODE REVIEW

File: Resources.php [hunk 1]
```diff
@@ -651,6 +661,16 @@
 		'position' => 'top',
 	),
 
+	'ext.srf.gallery.lightgallery' => $formatModule + array(
+		'scripts' => 'gallery/resources/ext.srf.gallery.lightgallery.js',
+		'styles'  => 'gallery/resources/ext.srf.gallery.lightgallery.css',
+		'dependencies' => array(
+			'ext.srf.formats.gallery',
+			'ext.jquery.lightgallery'
```
Comment (JeroenDeDauw) [inline Resources.php]: So these new files are always loaded even when the feature is not used? That's not ideal

File: formats/gallery/Gallery.php [hunk 2]
```diff
@@ -316,7 +316,18 @@ protected function addImageToGallery( ImageGallery &$ig, Title $imgTitle, $imgCa
...
 		$imgAlt =  $this->params['redirects'] === '' ? $imgCaption : $imgRedirect !== '' ? $imgRedirect : '' ;
-		$ig->add( $imgTitle, $imgCaption, $imgAlt );
+
+		if ($this->params['lightgallery']) {
+			$file = wfFindFile( $imgTitle );
+			if ( $file && $file->exists() ) {
+				$url = $file->getUrl();
+				$ig->add( $imgTitle, $imgCaption, $imgAlt, $url );
+			} else { // Not sure what to do here
+				// $ig->add( $imgTitle, $imgCaption, $imgAlt );
```
Comment (JeroenDeDauw) [inline formats/gallery/Gallery.php]: This does not look encouraging...

File: formats/gallery/Gallery.php [hunk 3]
```diff
@@ -316,7 +316,18 @@ protected function addImageToGallery( ImageGallery &$ig, Title $imgTitle, $imgCa
 		}
 		// Use image alt as helper for either text
 		$imgAlt =  $this->params['redirects'] === '' ? $imgCaption : $imgRedirect !== '' ? $imgRedirect : '' ;
-		$ig->add( $imgTitle, $imgCaption, $imgAlt );
+
+		if ($this->params['lightgallery']) {
```
Comment (JeroenDeDauw) [inline formats/gallery/Gallery.php]: Missing spaces

File: formats/gallery/Gallery.php [hunk 4]
```diff
@@ -316,7 +316,18 @@ protected function addImageToGallery( ImageGallery &$ig, Title $imgTitle, $imgCa
...
+			$file = wfFindFile( $imgTitle );
+			if ( $file && $file->exists() ) {
+				$url = $file->getUrl();
+				$ig->add( $imgTitle, $imgCaption, $imgAlt, $url );
+			} else { // Not sure what to do here
+				// $ig->add( $imgTitle, $imgCaption, $imgAlt );
+			}
+		} else {
+			$ig->add( $imgTitle, $imgCaption, $imgAlt );
+		}
```
Comment (JeroenDeDauw) [inline formats/gallery/Gallery.php]: This function contains a lot of if-else... perhaps something can be split out of it?

File: formats/gallery/resources/ext.srf.gallery.lightgallery.css [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * @file
```
Comment (JeroenDeDauw) [inline formats/gallery/resources/ext.srf.gallery.lightgallery.css]: No really? :)

File: formats/gallery/resources/ext.srf.gallery.lightgallery.css [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @since 2.4.2
+ * @revision 0.1
+ *
+ * @ingroup SRF
+ *
+ * @license GNU GPL v2+
+ * @author nischayn22
+ *
+ * @ignore
```
Comment (JeroenDeDauw) [inline formats/gallery/resources/ext.srf.gallery.lightgallery.css]: Ignore what?

File: formats/gallery/resources/ext.srf.gallery.lightgallery.css [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * @file
+ *
+ * @since 2.4.2
+ * @revision 0.1
```
Comment (JeroenDeDauw) [inline formats/gallery/resources/ext.srf.gallery.lightgallery.css]: Not correct any more, and not sure how this is useful.

**Comment by JeroenDeDauw** [comment]
Some superficial remarks from my side

**Comment by nischayn22** [comment]
Do we also need tests for this?

**Comment by JeroenDeDauw** [comment]
testall

**Comment by nischayn22** [comment]
My bad. I will retry when I have time. Thanks for your time and help reviewing.

**Pull request closed without merging**